script/tool/lib/src/version_check_command.dart

```diff
@@ -200,27 +206,73 @@ class VersionCheckCommand extends PackageLoopingCommand {
 
     final List<String> errors = <String>[];
 
-    bool versionChanged;
+    final CiConfig? ciConfig = package.parseCiConfig();
     final _CurrentVersionState versionState =
         await _getVersionState(package, pubspec: pubspec);
-    switch (versionState) {
-      case _CurrentVersionState.unchanged:
-        versionChanged = false;
-      case _CurrentVersionState.validIncrease:
-      case _CurrentVersionState.validRevert:
-      case _CurrentVersionState.newPackage:
-        versionChanged = true;
-      case _CurrentVersionState.invalidChange:
-        versionChanged = true;
-        errors.add('Disallowed version change.');
-      case _CurrentVersionState.unknown:
-        versionChanged = false;
-        errors.add('Unable to determine previous version.');
-    }
-
-    if (!(await _validateChangelogVersion(package,
-        pubspec: pubspec, pubspecVersionState: versionState))) {
-      errors.add('CHANGELOG.md failed validation.');
+    final bool usesBatchRelease = ciConfig?.isBatchRelease ?? false;
+    // PR with post release label is going to sync changelog.md and pubspec.yaml
+    // change back to main branch. We can proceed with ragular version check.
+    final bool hasPostReleaseLabel =
+        _prLabels.contains('post-release-${pubspec.name}');
+    bool versionChanged = false;
```

Comment:
Why did you add pre-initialization of this variable? This removes compiler validation that we always remember to set this in every code path.